[Gavin Cho] iP

Gavzzz opened the pull request.

Commit:
Add Gradle support

Gavzzz committed:
Level 1 implemented

Gavzzz committed:
Level 2 implemented

Gavzzz committed:
edited Level 2

Gavzzz committed:
Level 3 implemented

Gavzzz committed:
Task class implemented

Gavzzz committed:
Level 3 edited

Gavzzz committed:
Level 4 implemented

Gavzzz committed:
Level 5 implemented

Gavzzz committed:
Level 6 implemented

rycs2812 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
+
+    protected String by;
```
I think it might be better if the String had a more descriptive name, perhaps something like "date" or "deadline"?

rycs2812 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,130 @@
+
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();
+
+
+
```
I think it might be better if there was only a single blank line here

rycs2812 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,130 @@
+            if (input.equals("list")) {
+                System.out.println("___________________________________");
+                for (int i = 0; i < counter; i++) {
+                    System.out.println( (i+1) + "."
```
Seems like there's an extra whitespace on this line, might need to remove it to follow the coding standards

rycs2812 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,130 @@
+
+            else if (input.equals("mark")) {
+                int number = sc.nextInt();
+                collection.get(number-1).mark();
```
Similar issue on missing/extra whitespaces, noticed this issue in several other places too, might have to address these to follow the coding standards

rycs2812 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,130 @@
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("___________________________________");
```
Perhaps the line break could be stored in a constant since it is used multiple times in the code

rycs2812 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,27 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
Good job on following the coding standards for naming boolean variables👍

rycs2812 reviewed (commented):
Apart from the styling issues, I think the code is generally pretty okay. As long as all the increments are implemented eventually, I think the project will turn out fine!

Gavzzz committed:
Level 7 implemented

ssyap98 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,201 @@
+                Constants constant = Constants.valueOf(reader.next().toUpperCase());
+                Task task;
+                switch (constant) {
+                    case TODO:
```
There should not be indentation for the case clauses. It should look something like this
`switch (constant) {`
 `case TODO:`

ssyap98 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,22 @@
+
+public class Event extends Task{
+
+    protected String at;
```
Maybe you can change the variable name from "at" to "eventTime" so that it sounds more meaningful?

ssyap98 reviewed (commented):
Overall code looks good! Just have to make some minor changes to follow the java coding standard!

Gavzzz committed:
Level-8 implemented

Gavzzz committed:
More oop implemented

Gavzzz committed:
Packages enabled

Gavzzz committed:
Merge branch 'add-gradle-support'

Gavzzz committed:
Level 9 implemented

Gavzzz committed:
updated level 8

Gavzzz committed:
updated level 9

Gavzzz committed:
Merge branch 'branch-Level-9'

Gavzzz committed:
updated level 8

Gavzzz committed:
Merge branch 'branch-Level-8'

Gavzzz committed:
java docs implemented

Gavzzz committed:
Coding standard implemented

Gavzzz committed:
JUnit Tests implemented

Gavzzz committed:
Jar file dependencies implemented

Gavzzz committed:
Implemented GUI

location not found error when gui

Gavzzz committed:
Assertions implemented

Gavzzz committed:
implemented code quality checking

Gavzzz committed:
Update README.md

Gavzzz committed:
Detect duplicates implemented

Detecting duplicates method implemented.

As a user I want to know if my planner has duplicates and if I accidentally added duplicates.

Hence, this method is important

Gavzzz committed:
Merge branch 'branch-BCD-Extension'

Gavzzz committed:
Fixing level 8 branch issue and detect duplicates

Gavzzz committed:
GUI bug fixed

Changed return type to String so return type can be more easily accessed  by the GUI.

Gavzzz committed:
User guide implemented

Gavzzz committed:
Update README.md

Gavzzz committed:
Update README.md

Gavzzz committed:
Update README.md

Gavzzz committed:
Removed unnecessary imports

Gavzzz committed:
Merge branch 'master' of https://github.com/Gavzzz/ip

Gavzzz committed:
Fix Jar File unable to open issue and add new bcd tags extension

Removed command package to fix some issues regarding the Jar FIle

Gavzzz committed:
fix test file naming
